core/src/dunit/scala/io/snappydata/cluster/SplitClusterDUnitTest.scala

```diff
@@ -209,6 +209,13 @@ object SplitClusterDUnitTest extends SplitClusterDUnitTestObject {
       selectFromTableUsingJDBC("splitModeTable1", 1005, stmt)
     }
 
+    // check for SNAP-2156/2164
+    val updateSql = s"update splitModeTable1 set col1 = 100 where exists " +
+        s"(select 1 from splitModeTable1 t where t.col1 = splitModeTable1.col1 and t.col1 = 1234)"
+    assert(!stmt.execute(updateSql))
+    assert(stmt.getUpdateCount == 0)
+    assert(stmt.executeUpdate(updateSql) == 0)
```

Comment:
We would have tests also where return is positive count too?

Reply:
Ok, though the issue was about these operations throwing exceptions. Updates being proper is tested in details elsewhere like ColumnUpdateDeleteTests.